### DISCUSSION TITLE
state of cypher support / label expressions

### DISCUSSION THREAD

**Discussion by georgfaust** [FalkorDB/FalkorDB#457] [Q&A] [answered]
for this query

```
"MATCH (n:Movie|Person) RETURN n"
```

I get back `Invalid input '|': expected a label, '{', a parameter or ')'`

while it is a valid label expression:

```
labelExpression ::= ":" labelTerm

labelTerm ::=
    labelIdentifier
  | labelTerm "&" labelTerm
  | labelTerm "|" labelTerm
  | "!" labelTerm
  | "%"
  | "(" labelTerm ")"
```
https://neo4j.com/docs/cypher-manual/current/patterns/reference/#label-expressions-syntax

**Comment by gkorland** [comment]
You're right this syntax is still not supported by FalkorDB I opened an issues: https://github.com/FalkorDB/FalkorDB/issues/458

Meanwhile you can workaround it in different ways:

`MATCH (n) WHERE [] <> [x IN ['Movie', 'Person'] WHERE x in labels(n) | x] return n`
Or
`MATCH (n) WHERE 'Movie' in labels(n) OR 'Person' in labels(n) return n`

**Reply by georgfaust** [reply]
OK, is there a complete overview of supported cypher features - as this is not mentioned here:

https://docs.falkordb.com/cypher_support.html

But it states "Patterns are fully supported."

**Reply by gkorland** [reply]
you're right we'll extend the docs on this too

**Reply by AviAvni** [reply, marked as answer]
you are looking at the neo4j documentation and this feature is not part of the OpenCypher
see https://s3.amazonaws.com/artifacts.opencypher.org/M23/railroad/NodePattern.html#NodeLabels
sure we can still support this in the future

**Reply by georgfaust** [reply]
indeed, I need to have a look at openCypher, just assumed it implements all the feature the reference does. At least the OR-label-expression would be very useful.
